drivers/radio/wcn14a2a.c

```diff
@@ -0,0 +1,1139 @@
+/*
+ * Copyright (c) 2018 Open Source Foundries Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "radio_wcn14a2a"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_RADIO_LEVEL
+
+#include <logging/sys_log.h>
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <gpio.h>
+#include <device.h>
+#include <init.h>
+
+#include <console/uart_pipe.h>
+#include <net/net_offload.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+#if defined(CONFIG_NET_UDP)
+#include <net/udp.h>
+#endif
+#if defined(CONFIG_NET_TCP)
+#include "../../subsys/net/ip/tcp.h"
```

Comment:
It looks like if we should probably move the TCP header manipulation routines to public place so we could avoid these kind of hacks. Can you separate the needed functions from this file and create new one in include/net? We could then rename the subsys/net/ip/tcp.h -> subsys/net/ip/tcp_internal.h (so similar way as udp is doing things).